src/components/pre-search/manage-pre-search.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <div v-if="!$x.query.searchBox">
+    <h1 v-if="!hasQueryPreviews" class="x-title1 max-desktop:x-title1-sm max-desktop:x-px-16">
+      {{ $t('popularSearches.title') }}
+    </h1>
+    <CustomQueryPreview :queriesPreviewInfo="infoToRender" />
+  </div>
+</template>
+
+<script lang="ts">
+  import { computed, defineComponent, inject } from 'vue';
+  import { QueryPreviewInfo } from '@empathyco/x-components/queries-preview';
+
+  import { PopularSearchesState } from '@empathyco/x-components/types';
+  import { XPlugin } from '@empathyco/x-components';
+  import { popularSearchesXModule } from '@empathyco/x-components/popular-searches';
+  import { useStore } from '../../composables/use-store.composable';
+  import CustomQueryPreview from './custom-query-preview.vue';
+
+  export default defineComponent({
+    components: {
+      CustomQueryPreview
+    },
+    props: {
+      maxPopularSearchesToRender: {
+        type: Number,
+        default: 4
+      }
+    },
+    setup(props) {
+      const injectedQueriesPreviewInfo = computed<QueryPreviewInfo[]>(() => {
+        const injectedQueriesPreview = inject<QueryPreviewInfo[] | { value: QueryPreviewInfo[] }>(
+          'queriesPreviewInfo',
+          []
+        );
+        return 'value' in injectedQueriesPreview
+          ? injectedQueriesPreview.value
+          : injectedQueriesPreview;
+      });
+
+      const hasQueryPreviews = computed<boolean>(
+        () => injectedQueriesPreviewInfo.value.length !== 0
+      );
+
+      const popularSearchesAsQueryPreviewInfo = (): QueryPreviewInfo[] => {
+        XPlugin.registerXModule(popularSearchesXModule);
```

Comment:
Also, if you convert this to a computed property, do this directly in setup. Ideally this should be called once if the component has to use the popular searches, but it's not critical